PayloadPkg/FirmwareUpdate/FirmwareUpdateHelper.c

```diff
@@ -590,6 +591,67 @@ IsUpdateComponentForContainer (
    return TRUE;
 }
 
+/**
+  Perform Config data svn check
+
+  This function will perform svn checks for cfgdata in flash and
+  config data in capsule.
+
+  @param[in]  ImageHdr       Pointer to fw mgmt capsule Image header
+  @param[in]  FwPolicy       Firmware update policy
+  @param[out] SvnStatus      Svn compare status
+
+  @retval  EFI_SUCCESS      SVN check successful.
+  @retval  other            error occurred during firmware update
+**/
+EFI_STATUS
+CheckSblConfigDataSvn (
+  IN   EFI_FW_MGMT_CAP_IMAGE_HEADER  *ImageHdr,
+  IN   FIRMWARE_UPDATE_POLICY         FwPolicy,
+  OUT  UINT8                         *SvnStatus
+  )
+{
+  CDATA_BLOB_ON_FLASH       *CfgBlobCapAddr;
+  CDATA_BLOB_ON_FLASH       *CfgBlobFlashDataPtr;
+  UINT32                     CfgBlobFlashAddr;
+  UINT32                     CfgBlobSize;
+  EFI_STATUS                 Status;
+
+  //
+  // Get base address of CFGDATA from  firmware block to update
+  //
+  Status = EFI_INVALID_PARAMETER;
+
+  if (FwPolicy.Fields.UpdatePartitionB == 0x1) {
+    Status = GetComponentInfoByPartition(FLASH_MAP_SIG_CFGDATA, TRUE, (UINT32 *) &CfgBlobFlashAddr, &CfgBlobSize);
+  } else if (FwPolicy.Fields.UpdatePartitionA == 0x1) {
+    Status = GetComponentInfoByPartition(FLASH_MAP_SIG_CFGDATA, FALSE, (UINT32 *) &CfgBlobFlashAddr, &CfgBlobSize);
+  }
+
+  if (EFI_ERROR(Status)) {
+    DEBUG((DEBUG_ERROR, "Error in retrieving config data on flash: %r\n", Status));
+    return Status;
+  }
+
+  CfgBlobFlashDataPtr = (CDATA_BLOB_ON_FLASH *) (UINTN) CfgBlobFlashAddr;
+
+  // Locate config data blob header info from capsule image
+  CfgBlobCapAddr = (CDATA_BLOB_ON_FLASH *)((UINTN)ImageHdr + sizeof(EFI_FW_MGMT_CAP_IMAGE_HEADER));
+  if (CfgBlobCapAddr->Signature != CFG_DATA_SIGNATURE){
+    DEBUG((DEBUG_INFO, "Config data signature in capsule image does not match\n"));
+    return EFI_NOT_FOUND;
+  }
+
+  //Check SVN
+  if (CfgBlobCapAddr->Svn >= CfgBlobFlashDataPtr->Svn){
+    *SvnStatus = 1;

```

Comment:
Add checking for SvnStatus is NULL or not.

Reply:
Added SvnStatus check at start of function in Ln: 622.